### PR TITLE
Adds Cypress for local end-to-end testing

### PR DESCRIPTION
**Pull request by Kongkille**
Adds cypress to the project allowing for end-to-end testing.

Tests can be run with 

`yarn run cypress-test`

To complete all tests, a snowpack token is required (emap internal team) but the onboarding tests can be completed without any authorization.

The idea is that this should be able to be invoked automatically by a Github action when we're updating the app to make sure  things work!

Can be reviewed per commit

### CODE REVIEW

File: web/cypress/integration/countrypanel.spec.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable no-undef */
```
Comment (Kongkille) [inline web/cypress/integration/countrypanel.spec.js:1]: @madsnedergaard any tips for how to use the cypress keywords without disabling eslint rules?

Comment (madsnedergaard) [reply web/cypress/integration/countrypanel.spec.js:1]: We can either add the keywords to GLOBALS in eslint config, but we can also just add this plugin :)
https://www.npmjs.com/package/eslint-plugin-cypress

**Review by madsnedergaard — COMMENTED** [review]
This is awesome!! 🤩 

What's best practice here around the tests - is it worth having one "big" test that starts on frontpage, clicks a country, asserts country panel, etc. vs many smaller ones?
Guess it depends on how much it runs in parallel :)

I'm also super excited to start writing tests now!!

File: web/package.json [hunk 2]
```diff
@@ -69,6 +69,7 @@
     "colors": "^1.3.2",
```
Comment (madsnedergaard) [inline web/package.json:69]: yarn.lock should be committed too :)
```
error Your lockfile needs to be updated, but yarn was run with `--frozen-lockfile`.
```

File: web/package.json [hunk 3]
```diff
@@ -91,6 +92,7 @@
     "sass-loader": "10.1.1",
     "shapefile": "^0.6.2",
     "snowpack": "^3.8.8",
+    "topojson-client": "^3.1.0",
```
[the remark this answers is not in the document]
Comment (Kongkille) [reply web/package.json:95]: Oh this was part of getting the project to run, I think my snowpack somehow expected 3.1

Not sure why, I'll just remove it again

File: web/cypress.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "projectId": "9z8tgk",
+  "video": false,
+  "screenshotOnRunFailure": false
```
[the remark this answers is not in the document]
Comment (Kongkille) [reply web/cypress.json:4]: Makes sense!

File: web/cypress/integration/countrypanel.spec.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+describe('countryPanel displays data', () => {
+  it('asserts countryPanel contains carbon intensity when available', () => {
+    cy.visit('localhost:8080/zone/SE?remote=true');
```
[the remark this answers is not in the document]
Comment (Kongkille) [reply web/cypress/integration/countrypanel.spec.js:4]: Yeah I gave it some thought. Basically, I don't think the mockserver represents the environment well enough to actually qualify as end-to-end testing as of right now. 

However, not using fixed data is also a problem as the test is no longer isolated and could depend on incoming data. 

Ideally we'd want a mockserver replicating end environment as much as possible, and using that here would help us incentivize keeping it up to date. 

Perhaps we can use both? That would provide a very actionable guideline for when to update the mockserver if it fails a test that production does not and vice versa.

**Comment by Kongkille** [comment]
> This is awesome!! 🤩
> 
> What's best practice here around the tests - is it worth having one "big" test that starts on frontpage, clicks a country, asserts country panel, etc. vs many smaller ones? Guess it depends on how much it runs in parallel :)
> 
> I'm also super excited to start writing tests now!!

Well I looked a bit into best practices, based on this [talk](https://www.youtube.com/watch?v=5XQOK0v_YRE&ab_channel=OKG%21-PaulDowman%27stecheventsandinterviews) which suggests organizing tests into features but it does not seem completely obvious to me how to relate it to our case. 

However looking further into a concrete example (they maintain some best-practice examples) it seems that a larger test that has several "it" statements is more correct.

https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui/bankaccounts.spec.ts

**Comment by VIKTORVAV99** [comment]
Does it make sense to move the spec files to the same place as their respective modules/components?
It would make it a little easier find IMO and with the new [explorer file nesting](https://code.visualstudio.com/updates/v1_67#_explorer-file-nesting) in vs code it wouldn't be cluttered (if someone use vs code that is).

File: web/cypress/integration/countrypanel.spec.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (madsnedergaard) [reply web/cypress/integration/countrypanel.spec.js]: Yeah, good point :)

**Comment by Kongkille** [comment]
@VIKTORVAV99 Right now I'm just following the Cypress best practices. I do think they'll be a bit hard to find, it will make sense for unit testing, but I am not so sure it will make sense for integration tests as it may test part of the code that is not in that specific folder

File: web/cypress/integration/onboarding.spec.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+describe('Onboarding appears for the first time', () => {
```
Comment (Kongkille) [inline web/cypress/integration/onboarding.spec.js:1]:
```suggestion
describe('Onboarding', () => {
```

**Review by madsnedergaard — APPROVED** [review]

**Comment by VIKTORVAV99** [comment]
@Kongkille makes sense! 👍🏻

**Pull request merged**